src/Symfony/Component/HttpFoundation/Tests/RequestTest.php

```diff
@@ -2427,6 +2427,18 @@ public function testTrustedPort()
 
         $this->assertSame(443, $request->getPort());
     }
+
+    public function testTrustedPortDoesNotDefaultToZero()
+    {
+        Request::setTrustedProxies(['1.1.1.1'], Request::HEADER_X_FORWARDED_ALL);
+
+        $request = Request::create('/');
+        $request->server->set('REMOTE_ADDR', '1.1.1.1');
+        $request->headers->set('X-Forwarded-Host', 'test.example.com');
+        $request->headers->set('X-Forwarded-Port', null);
```

Comment:
HeaderBag::set does not accept null. Maybe you meant to use an empty string.

Reply:
bd498f2503